src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -328,8 +484,317 @@ get_service_ip(sd_bus_message *m,
     } while (1);
 
 end:
-    if (r == 0)
+    exit = sd_bus_message_exit_container(m);
+    SOL_INT_CHECK(exit, < 0, exit);
+
+    return r;
+}
+
+static int
+get_proxy_info(sd_bus_message *m, struct sol_proxy *proxy)
+{
+    char *str;
+    uint16_t i;
+    int r, exit;
+
+    r = sd_bus_message_enter_container(m, SD_BUS_TYPE_ARRAY, "{sv}");
+    SOL_INT_CHECK(r, < 0, r);
+
+    do {
+        r = sd_bus_message_enter_container(m, SD_BUS_TYPE_DICT_ENTRY, "sv");
+        SOL_INT_CHECK(r, < 0, r);
+
+        r = sd_bus_message_read_basic(m, SD_BUS_TYPE_STRING, &str);
+        SOL_INT_CHECK_GOTO(r, < 0, end);
+        if (streq(str, "Method")) {
+            char *method;
+
+            static const struct sol_str_table table[] = {
+                SOL_STR_TABLE_ITEM("direct",
+                    SOL_PROXY_METHOD_DIRECT),
+                SOL_STR_TABLE_ITEM("auto",
+                    SOL_PROXY_METHOD_AUTO),
+                SOL_STR_TABLE_ITEM("manual",
+                    SOL_PROXY_METHOD_MANUAL),
+                { }
+            };
+
+            r = sd_bus_message_enter_container(m, SD_BUS_TYPE_VARIANT, "s");
+            SOL_INT_CHECK(r, < 0, r);
+
+            r = sd_bus_message_read_basic(m, SD_BUS_TYPE_STRING, &method);
+            SOL_INT_CHECK_GOTO(r, < 0, end);
+
+            if (method)
```

Comment:
it is possible ,  but according to API spec , there is no reference to treat as an error.
please check: https://www.freedesktop.org/software/systemd/man/sd_bus_message_read_basic.html